### PR TITLE
Initial push

### PR DESCRIPTION
**Pull request by Jamie-BitFlight**
## what
* Implement SNS alerts from cloudwatch for EFS

## why
* Automate monitoring infra

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,49 @@
-# terraform-aws-efs-cloudwatch-sns-alarms
\ No newline at end of file
+# terraform-aws-rds-alerts
```
Comment (osterman) [inline README.md]: Update name

File: README.md [hunk 2]
```diff
@@ -1 +1,49 @@
...
+  name                 = "mydb"
+  username             = "foo"
+  password             = "foobarbaz"
+  parameter_group_name = "default.mysql5.7"
+  apply_immediately = "true"
+  skip_final_snapshot = "true"
+}
+
+module "rds_alarms" {
+  source = "github::https://github.com/bitflight-public/terraform-aws-rds-alerts.git?ref=master"
```
Comment (osterman) [inline README.md]: Update URL

File: examples/marbot/main.tf [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  protocol  = "https"
+  endpoint  = "https://api.marbot.io/v1/endpoint/${var.marbot_endpoint_id}"
+}
+
+resource "aws_efs_file_system" "default" {
+  creation_token = "app"
+}
+
+module "alarms" {
+  source         = "github::https://github.com/bitflight-public/terraform-aws-rds-alerts.git?ref=master"
```
Comment (osterman) [inline examples/marbot/main.tf]: Update url

File: .gitignore [hunk 4]
```diff
@@ -1,9 +1,9 @@
...
 *.tfstate
 *.tfstate.*
+**/*.pem
+**/.terraform
+**/terraform.tfstate
+**/terraform.tfstate.backup
 
-# .tfvars files
-*.tfvars
+dev.auto.tfvars
```
Comment (aknysh) [inline .gitignore]: this probably can be removed

File: README.md [hunk 5]
```diff
@@ -1 +1,49 @@
...
+  storage_type         = "gp2"
+  engine               = "mysql"
+  engine_version       = "5.7"
+  instance_class       = "db.t2.micro"
+  identifier_prefix    = "rds-server-example"
+  name                 = "mydb"
+  username             = "foo"
+  password             = "foobarbaz"
+  parameter_group_name = "default.mysql5.7"
+  apply_immediately = "true"
```
Comment (aknysh) [inline README.md]: `terraform fmt`

File: README.md [hunk 6]
```diff
@@ -1 +1,49 @@
-# terraform-aws-efs-cloudwatch-sns-alarms
\ No newline at end of file
+# terraform-aws-rds-alerts
+
+Create a set of sane RDS CloudWatch alerts for monitoring the health of an RDS instance.
```
Comment (aknysh) [inline README.md]: This should describe `EFS`, not `RDS`

File: README.md [hunk 7]
```diff
@@ -1 +1,49 @@
...
+| CPU     | CPUUtilization   | `>`                  | 80 %      | Queuing theory tells us the latency increases exponentially with utilization. In practice, we see higher latency when utilization exceeds 80% and unacceptable high latency with utilization above 90% |
+| CPU     | CPUCreditBalance | `<`                  | 20        | One credit equals 1 minute of 100% usage of a vCPU. 20 credits should give you enough time to a) fix the inefficiency, b) add capacity or c) don't use t2 type.                                        |
+| Memory  | FreeableMemory   | `<`                  | 64 MB     | This number is calculated from our experience with RDS workloads.                                                                                                                                      |
+| Memory  | SwapUsage        | `>`                  | 256 MB    | Sometimes you can not entirely avoid swapping. But once the database accesses paged memory, it will slow down.                                                                                         |
+
+## Example
+
+```hcl
+resource "aws_db_instance" "default" {
```
Comment (aknysh) [inline README.md]: This should describe `EFS`, not `RDS`

File: README.md [hunk 8]
```diff
@@ -1 +1,49 @@
...
+  identifier_prefix    = "rds-server-example"
+  name                 = "mydb"
+  username             = "foo"
+  password             = "foobarbaz"
+  parameter_group_name = "default.mysql5.7"
+  apply_immediately = "true"
+  skip_final_snapshot = "true"
+}
+
+module "rds_alarms" {
```
Comment (aknysh) [inline README.md]: This should describe `EFS`, not `RDS`

File: README.md [hunk 9]
```diff
@@ -1 +1,49 @@
...
+  source = "github::https://github.com/bitflight-public/terraform-aws-rds-alerts.git?ref=master"
+  db_instance_id = "${aws_db_instance.default.id}"
+}
+```
+
+## Variables
+| Name                         | Description                                                                              | Required |
+|------------------------------|------------------------------------------------------------------------------------------|----------|
+| db_instance_id               | The instance ID of the RDS database instance that you want to monitor.               		| Yes 		 |
```
Comment (aknysh) [inline README.md]: This should describe `EFS`, not `RDS`

**Comment by Jamie-BitFlight** [comment]
@aknysh I have reviewed your changes and applied them.
After the pull request has been accepted, I think the repo needs to be renamed: 
**terraform-aws-cloudwatch-alarms-efs**

**Comment by osterman** [comment]
Ok! We should have some time this week to review. Thanks @Jamie-BitFlight

File: README.md [hunk 10]
```diff
@@ -1 +1,30 @@
...
+module "efs_alarms" {
+  source         = "../../"
+  filesystem_id = "${aws_efs_file_system.default.id}"
+}
+```
+
+## Variables
+| Name                           | Description                                                                               | Required |
+|--------------------------------|-------------------------------------------------------------------------------------------|----------|
+| filesystem_id                  | The instance ID of the RDS database instance that you want to monitor.               		 | Yes 		 |
```
Comment (osterman) [inline README.md]: This should be `EFS`

**Pull request merged**